tests/integration/cocoon/test_replace_current_transaction_alias.py

```diff
@@ -0,0 +1,134 @@
+
+
+class CocoonTestTransactionTypeUpload(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+
+        cls.system_configuration_api = cls.api_factory.build(
+            lusid.api.SystemConfigurationApi
+        )
+
+        cls.alias = models.TransactionConfigurationTypeAlias(
+            type=new_transaction_type,
+            description="TESTBUY1",
+            transaction_class="TESTBUY1",
+            transaction_group="SYSTEM1",
+            transaction_roles="AllRoles",
+        )
+
+        cls.movements = [
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="StockMovement",
+                side="Side1",
+                direction=1,
+                properties={},
+                mappings=[],
+            ),
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="CashCommitment",
+                side="Side2",
+                direction=-1,
+                properties={},
+                mappings=[],
+            ),
+        ]
+
+        cls.create_transaction_response = cls.system_configuration_api.create_configuration_transaction_type(
+            transaction_configuration_data_request=models.TransactionConfigurationDataRequest(
+                aliases=[cls.alias], movements=cls.movements
+            )
+        )
+
+    def test_update_current_alias_with_new_movements(self):
+
+        updated_movements = [
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="StockMovement",
+                side="Side1",
+                direction=-1,
+                properties={},
+                mappings=[],
+            ),
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="CashCommitment",
+                side="Side2",
+                direction=1,
+                properties={},
+                mappings=[],
+            ),
+        ]
+
+        replace_current_transaction_alias(
+            api_factory=self.api_factory,
+            updated_alias=self.alias,
+            movement=updated_movements,
+        )
+
+        get_transaction_types = (
+            self.system_configuration_api.list_configuration_transaction_types()
+        )
+
+        uploaded_alias = []
+
+        for trans_type in get_transaction_types.transaction_configs:
+            for alias in trans_type.aliases:
+                if alias.type == new_transaction_type:
+                    uploaded_alias.append(trans_type)
+
+        alias_for_test = models.TransactionConfigurationData(
+            aliases=[self.alias], movements=updated_movements, properties={}
+        )
+        self.assertEqual(uploaded_alias[0].to_dict(), alias_for_test.to_dict())
```

Comment:
Does this assertion fail if you don't convert them to_dict()? You can see [here](https://github.com/finbourne/lusid-sdk-python-preview/blob/master/sdk/lusid/models/transaction_configuration_data.py) in the __eq__ function -  that when you check that these two classes are equal it will check they are of the same class than to a to_dict conversion anyway.

Reply:
It did fail originally as I was using a **_TransactionConfigurationMovementDataRequest_** rather than a **_TransactionConfigurationMovementData_**.
I updated the model and the eq method gives correct results.